src/main/java/mate/academy/springboot/swagger/controller/ProductController.java

```diff
@@ -0,0 +1,101 @@
+
+@RestController
+@RequestMapping("/products")
+public class ProductController {
+    private final ProductService productService;
+    private final ProductMapper productMapper;
+    private final PageRequestParser pageRequestParser;
+
+    public ProductController(ProductService productService,
+                             ProductMapper productMapper,
+                             PageRequestParser pageRequestParser) {
+        this.productService = productService;
+        this.productMapper = productMapper;
+        this.pageRequestParser = pageRequestParser;
+    }
+
+    @PostMapping
+    @ApiOperation(value = "create a new product")
+    public ProductResponseDto create(@RequestBody ProductRequestDto dto) {
+        Product savedProduct = productService.save(productMapper.toModel(dto));
+        return productMapper.toDto(savedProduct);
+    }
+
+    @GetMapping("/{id}")
+    @ApiOperation(value = "get a product by id")
+    public ProductResponseDto getById(@PathVariable Long id) {
+        return productMapper.toDto(productService.findById(id));
+    }
+
+    @DeleteMapping("/{id}")
+    @ApiOperation(value = "delete a product by id")
+    public void deleteById(@PathVariable Long id) {
+        productService.deleteById(id);
+    }
+
+    @PutMapping("/{id}")
+    @ApiOperation(value = "update a product by id")
+    public ProductResponseDto update(@PathVariable Long id,
+                                     @RequestBody ProductRequestDto dto) {
+        Product product = productMapper.toModel(dto);
+        product.setId(id);
+        return productMapper.toDto(productService.update(product));
+    }
+
+    @GetMapping
+    @ApiOperation(value = "get a list of sorted products (ASC order by default "
+            + "with pagination and filtering by price options. "
+            + "To sort by field in ASC or DESC order, type 'fieldName':'order'.")
+    public List<ProductResponseDto> findAll(@RequestParam (defaultValue = "0")
+            @ApiParam(value = "default value is 0") Integer page,
+                                            @RequestParam (defaultValue = "20")
+            @ApiParam(value = "default value is 20") Integer size,
+                                            @RequestParam (defaultValue = "id")
+            @ApiParam(value = "default value is id") String sortBy,
+                                            @RequestParam (defaultValue = "0")
+            @ApiParam(value = "default value is 0") BigDecimal from,
+                                            @RequestParam (defaultValue = "1000000")
+            @ApiParam(value = "default value is 1000000") BigDecimal to) {
+        PageRequest pageRequest = pageRequestParser.parse(page, size, sortBy);
+        return productService.findAllByPriceBetween(from, to, pageRequest).stream()
+                .map(productMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @GetMapping("/inject")
```

Comment:
I am not sure, should we left this method at final solution?

Reply:
you're right, not really =)